### DISCUSSION TITLE
Cleaned Activity for Game Entries

### DISCUSSION THREAD

**Discussion by SporyTike** [RetroAchievements/RAWeb#1092] [Ideas]
Since a while we now have the feature to see player activity for a specific set on the related game entry page. The general idea is to see who is currently playing the game and what their progress is. To do that the feature grabs the last activity of every player and filters which one played the game related to this specific game ID, then shows these on the game page.

We have hundrets of accounts on the site that were inactive for months or even years, some which likely will never have any activity again and they shouldn't be shown in the activity sections.
Taking this game entry as example: [https://retroachievements.org/game/2108](https://retroachievements.org/game/2108)
![image](https://user-images.githubusercontent.com/45672834/184870285-bf53137c-fa73-4be2-a830-33a99c8c0c34.png)

There are a couple of issues with this:
1) It shows outdated information, if it was a completely different game in the past or had a different RP and afterwards the user never loaded another game it shows this for all eternity.
2) Technically connected to it it's secrurity issue because if an RP has an insult or something other unwelcomed and a player loaded the game with that RP and then went inactive, the only way to remove that activity would be by making a database edit.
3) Due to showing all activities even those who were years ago the sections become longer and longer everytime a user gets inactive. The game above is a good example: From 10 users only 2 are actually actively playing the game and are from this year. I doubt anyone is actually interested to know who played the game a year ago and then left the site, it just makes the list unnecessarily longer.

My suggestion would be to directly exclude all activities in the section of each game as soon the activity is older than two weeks. If the player didn't play the game within two weeks, I wouldn't count it as active anymore and unrelevant information that doesn't need to be shown. The activity field only shows the very last gaming activity of a user and is mostly only relevant to see who is playing right now, no need to show history - for that people should check the profile of a specific user, recent mastery list or single achievement unlocks.

**Comment by Jamiras** [comment]
The feature is just using the "Last Seen In" information from the user profiles. If there's something outdated or offensive, it'll still be visible there. For that, we'd need an admin tool to clear it out.

One thing that exaggerates the problem is that as soon as a player moves to another game, their "Last Seen In" changes, so they're no longer a recent player of the current game. I think there's some DB changes in the v2 schema that would retain the player's last "Last Seen In" message for each game they've played. That would make more recent users appear in the list, bumping the older players out of sight.

I think two weeks might be too small of a window. But I could see limiting it to 6 months or a year.
